src/GrpAb/Map.jl

```diff
@@ -496,3 +497,28 @@ function hom(G::GrpAbFinGen, H::GrpAbFinGen; task::Symbol = :map)
   end
   return R, MapFromFunc(phi, ihp, R, MapParent(G, H, "homomorphisms"))
 end
+
+################################################################################
+#
+#  Pre- and postinverse
+#
+################################################################################
+
+function _prepostinverse(f::GrpAbFinGenMap)
+  # in the surjective case, we find the preimage and
+  # if it does not exist, we find any element of the domain,
+  # which is also fine
+  return [haspreimage(f, g)[2] for g in gens(codomain(f))]
+end
+
+# Given surjective f, find g such that fg = id
+function preinverse(f::GrpAbFinGenMap)
+  @req is_surjective(f) "Map must be surjective"
+  return hom(codomain(f), domain(f), _prepostinverse(f))
+end
+
+# Given surjective f, find g such that gf = id
+function postinverse(f::GrpAbFinGenMap)
+  @req is_injective(f) "Map must be injective"
+  return hom(codomain(f), domain(f), _prepostinverse(f))
+end
```

Comment:
both return the same? A more efficient version of pseudo_inverse? (well a linear version of pseud_inverse)?

Reply:
It is just by accident that we can do the "same" (because `haspreimage` returns two things). It is a more mathematical version of `pseudo_inverse`.